### PR TITLE
Restore FD position regarding multiple monitors.

### LINKED ISSUE #856: Window position (start on display #2) not remembered in my display configuration
FlashDevelop can't remember the last used monitor/position. My display settings are like this:
Settings
Display 1 is my laptop, and Display 2 is an external wall-mounted monitor, above my laptop. When I start FD, it always appears on laptop's screen, but I want it to start on the other screen. I tried un-maximizing FD window, moving it to screen 2, then restarting FD, then maximizing the window again. Most apps with "remember window size&position" feature usually remember which screen to start on after this procedure, but not in my case. Maybe it's related to my particular screen arrangement. Since screen 2 is above and to the left from screen 1, when FD is on screen 2, it's actual window position is -1920, -1080 (FlashDevelop Settings > State). Maybe the position doesn't get remembered or restored because negative coordinates are not handled correctly? For now as a work-around I have to use an AHK script which triggers after I launch FlashDevelop with a hotkey
```autohotkey
      WinWait, ahk_exe FlashDevelop.exe
      if WinExist("ahk_exe FlashDevelop.exe") {
         WinRestore
         WinMove -1920, -1080
         WinMaximize
         WinActivate
      }
```

### PR DESCRIPTION
**Pull request by wise0704**
Closes #856 

11 out of 14 now :)

### CODE REVIEW

**Comment by wise0704** [comment]
I'm gonna go to bed now... lol
Do I deserve a cookie??

**Comment by Meychi** [comment]
A gigantic one, thanks. :) I will check all these later today.

**Comment by Neverbirth** [comment]
The change seems to be valid, but I don't know if it's missing something else.

File: FlashDevelop/MainForm.cs [hunk 1]
```diff
@@ -903,10 +903,21 @@ public void CheckForUpdates()
         public void InitializeWindow()
         {
             this.WindowState = this.appSettings.WindowState;
-            Point position = new Point(this.appSettings.WindowPosition.X, this.appSettings.WindowPosition.Y);
-            if (position.X < -4) position.X = 0;
-            if (position.Y < -25) position.Y = 0;
-            this.Location = position;
+            Rectangle bounds = new Rectangle(this.appSettings.WindowPosition, this.appSettings.WindowSize);
+            Boolean validPosition = false;
```
Comment (wise0704) [inline FlashDevelop/MainForm.cs]: Maybe adding `bounds.Inflate(-4, -25)` here could be better, since previously it was like that.

**Comment by Meychi** [comment]
Soon as someone can test this, we can merge it.

**Comment by wise0704** [comment]
Just FYI: the code fragment is actually provided by Microsoft. Should be working.

**Comment by Meychi** [comment]
Ok fine for me then.

**Pull request merged**